lib/rom/sql/attribute.rb

```diff
@@ -14,6 +14,28 @@ class Attribute < ROM::Schema::Attribute
       # Error raised when an attribute cannot be qualified
       QualifyError = Class.new(StandardError)
 
+      TypeExtensions = {}
+
+      # Registers a set of operations supported for a specific type
+      #
+      # @example
+      #   ROM::SQL::Attribute.register_type(ROM::SQL::Types::PG::JSONB) do
+      #     def contains(type, keys)
+      #       Sequel::Postgres::JSONBOp.new(type.meta[:name]).contains(keys)
+      #     end
+      #   end
+      #
+      # @param [Dry::Types::Type] type Type
+      #
+      # @api public
+      def self.register_type(type, &block)
+        raise ArgumentError, "Type #{ type } already registered" if TypeExtensions.key?(type)
+        mod = Module.new(&block)
+        ctx = Object.new.extend(mod)
+        functions = mod.instance_methods.each_with_object({}) { |m, ms| ms[m] = ctx.method(m) }
+        TypeExtensions[type] = functions
```

Comment:
right, good point! 
BTW this way of extending is how ruby's dispatching works, but instead of having an object with methods we have a hash/list of methods directly. It's a bit more convenient because you have _only_ methods you defined, nothing leaks from `Object` and `Kernel`.